### PR TITLE
solution

### PR DESCRIPTION
**Pull request by igor-kashchenko**
[DEMO LINK](https://igor-kashchenko.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

**Review by OlhaYevstifieieva — APPROVED** [review]
Gj!

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
...
+    setSortType(SortType.LENGTH);
+  };
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const visibleGoods
+  = getReorderedGoods(goodsFromServer, { sortType, isReversed });
```
Comment (Anna-Tkachenko) [inline src/App.tsx:79]:
```suggestion
  const visibleGoods = getReorderedGoods(
    goodsFromServer,
    { sortType, isReversed },
  );
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
...
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={reverseGoods}
         >
           Reverse
         </button>
+        {(isReversed || sortType !== SortType.NONE)
```
Comment (Anna-Tkachenko) [inline src/App.tsx:113]: don't combine several checks directly in jsx
will be better to create separate boolean variable before the `return` statement and use this variable here to make code cleaner

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
-          className="button is-warning is-light"
+          className={classNames('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={reverseGoods}
         >
           Reverse
         </button>
+        {isVisible
+        && (
```
Comment (Anna-Tkachenko) [inline src/App.tsx:118]:
```suggestion
        {isVisible && (
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const isVisible = isReversed || sortType !== SortType.NONE;
```
Comment (Anna-Tkachenko) [inline src/App.tsx:61]: will be better to specify some noun here
for now it's not clear what should be visible or not

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**